larray/core/array.py

```diff
@@ -5420,10 +5420,16 @@ def divnot0(self, other):
             else:
                 return self / other
         else:
-            with np.errstate(divide='ignore', invalid='ignore'):
-                res = self / other
-            res[other == 0] = 0
-            return res
+            (self, other), res_axes = make_numpy_broadcastable((self, other))
+            otherdata = other.data
+            other_eq0 = otherdata == 0
+            # numpy array division gets slower the more zeros you have in other, so we change it before the division
+            # happens. This is obviously slower than doing nothing if we have very few zeros but I think it's a win
+            # on average given that other is likely to contain zeros when using divnot0.
+            otherdata = np.where(other_eq0, 1, otherdata)
+            res_data = self.data / otherdata
+            res_data[np.broadcast_to(other_eq0, res_data.shape)] = 0.0
```

Comment:
by assigning to 0.0, you implicitly convert the res_data to a float array. Am I right?